operator-framework-core/src/main/java/io/javaoperatorsdk/operator/processing/event/internal/CustomResourceEventSource.java

```diff
@@ -128,46 +135,16 @@ public void eventReceived(Watcher.Action action, T customResource) {
       return;
     }
 
-    if (!skipBecauseOfGeneration(customResource)) {
-      eventHandler.handleEvent(new CustomResourceEvent(action, customResource, this));
-      markLastGenerationProcessed(customResource);
-    } else {
+    if (!predicate.test(oldResource, customResource)) {
       log.debug(
           "Skipping event handling resource {} with version: {}",
           getUID(customResource),
           getVersion(customResource));
-    }
-  }
-
-  private void markLastGenerationProcessed(T resource) {
-    if (generationAware && resource.hasFinalizer(resourceFinalizer)) {
-      lastGenerationProcessedSuccessfully.put(
-          KubernetesResourceUtils.getUID(resource), resource.getMetadata().getGeneration());
-    }
-  }
 
-  private boolean skipBecauseOfGeneration(T customResource) {
-    if (!generationAware) {
-      return false;
-    }
-    // if CR being deleted generation is naturally not changing, so we process all the events
-    if (customResource.isMarkedForDeletion()) {
-      return false;
-    }
-
-    // only proceed if we haven't already seen this custom resource generation
-    Long lastGeneration =
-        lastGenerationProcessedSuccessfully.get(customResource.getMetadata().getUid());
-    if (lastGeneration == null) {
-      return false;
-    } else {
-      return customResource.getMetadata().getGeneration() <= lastGeneration;
+      return;
     }
-  }
 
-  @Override
-  public void eventSourceDeRegisteredForResource(String customResourceUid) {
```

Comment:
Why was this method removed?

Reply:
because the method was only about to remove the generation from the local cache, but since the cache does not exists anymore the method is useless (and the interface provide a default no-op method)